### PR TITLE
OSModifier: Add services and modules support in EMU API

### PR DESCRIPTION
**Pull request by elainezhao1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add services and modules support in EMU API

###### Change Log  

- Create a modulelist type
- Add functions to update services and modules in osmodifier

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Will work with trident crew to verify in trident-e2e pipeline

### CODE REVIEW

File: toolkit/tools/osmodifierapi/os.go
Comment (cwize1) [inline toolkit/tools/osmodifierapi/os.go:13]: It looks like the `toolkit/tools/osmodifierapi` directory isn't marked as being owned by the image customizer team in the [CODEOWNERS](https://github.com/microsoft/azurelinux/blob/3.0-dev/.github/CODEOWNERS) file. Hence why `CBL-Mariner-Stable-Maintainers` is a required reviewer.

**Review by cwize1 — APPROVED** [review]

File: toolkit/tools/osmodifierapi/os.go
Comment (elainezhao96) [reply toolkit/tools/osmodifierapi/os.go:13]: edited the file and it requires review from admin. I've contacted Jon to ask for approval.

**Review by PawelWMS — APPROVED** [review]
Approving the `CODEOWNERS` changes.

**Comment by elainezhao96** [comment]
close this as mic moved to its dedicated repo

**Pull request closed without merging**
